### PR TITLE
Add support for groups in Img2Col/Col2Img

### PR DESCRIPTION
**Pull request by bartekxk**

### CODE REVIEW

File: client_example/22_im2col_col2im/column_to_image.cpp [hunk 1]
```diff
@@ -52,18 +52,18 @@ int main()
     std::array<ck::index_t, 2> wei_spatial_lengths{Y, X};
     std::array<ck::index_t, 2> out_spatial_lengths{Ho, Wo};
 
-    // We have NHWGC in memory space (G is dummy)
+    // We have NHWGC in memory space
     // However, CK's API only accept length and stride with order of GNCHW
     // Hence, we need to adjust the order of stride
```
Comment (bwroblew) [inline client_example/22_im2col_col2im/column_to_image.cpp]:
```suggestion
    // However, CK's API only accepts lengths and strides with order of GNCHW.
    // Hence, we need to adjust the order of strides.
```

File: client_example/22_im2col_col2im/image_to_column.cpp [hunk 2]
```diff
@@ -52,19 +52,19 @@ int main()
     std::array<ck::index_t, 2> wei_spatial_lengths{Y, X};
     std::array<ck::index_t, 2> out_spatial_lengths{Ho, Wo};
 
-    // We have NHWGC in memory space (G is dummy)
+    // We have NHWGC in memory space
     // However, CK's API only accept length and stride with order of GNCHW
```
Comment (bwroblew) [inline client_example/22_im2col_col2im/image_to_column.cpp]: same

File: example/52_im2col_col2im/image_to_column_f32.cpp [hunk 3]
```diff
@@ -33,13 +33,13 @@ bool RunImageToColumn(const ExecutionConfig& config, const ck::utils::conv::Conv
...
         ck::utils::conv::make_input_host_tensor_descriptor_g_n_c_wis_packed<ImLayout>(conv_params);
-    const auto out_desc = HostTensorDescriptor({NDoHoWo, CZYX});
+    const auto out_desc = HostTensorDescriptor({G, NDoHoWo, CZYX});
 
     std::array<ck::index_t, NDimSpatial> input_spatial_lengths{};
     std::array<ck::index_t, NDimSpatial> filter_spatial_lengths{};
     std::array<ck::index_t, NDimSpatial> output_spatial_lengths{};
     std::array<ck::index_t, NDimSpatial + 3> image_g_n_c_wis_strides{};
-    std::array<ck::index_t, 2> gemm_m_k_strides{};
+    std::array<ck::index_t, 3> gemm_m_k_strides{};
```
Comment (bwroblew) [inline example/52_im2col_col2im/image_to_column_f32.cpp]: `g_m_k`

File: include/ck/tensor_operation/gpu/device/device_conv_tensor_rearrange.hpp [hunk 4]
```diff
@@ -14,11 +14,12 @@ namespace device {
 /**
  * \brief Convolution Tensor Rearrange.
  *
- * This Device operator supports conversion image ([G, N, Di, Hi, Wi, C]) to
- * the gemm problem([N * Do * Ho * Wo, Z *  Y * X * C]) (Image to Column) and
+ * This Device operator supports conversion image to
+ * the gemm problem (Image to Column) and
  * conversion gemm form to the image (Column to Image).
```
Comment (bwroblew) [inline include/ck/tensor_operation/gpu/device/device_conv_tensor_rearrange.hpp]:
```suggestion
 * This Device operator supports converting an image to
 * the GEMM representation (Image to Column) and
 * converting a GEMM form to the image (Column to Image).
```

File: library/include/ck/library/tensor_operation_instance/gpu/conv_tensor_rearrange.hpp [hunk 5]
```diff
@@ -19,109 +19,214 @@ namespace instance {
 
 using namespace ck::conv_tensor_rearrange_op;
 
+// GNWC/GNHWC/GNDHWC
 // Image to Column
 // nhwc, 1d
```
Comment (bwroblew) [inline library/include/ck/library/tensor_operation_instance/gpu/conv_tensor_rearrange.hpp]:
```suggestion
// GNHWC, 1d
```
same below

File: library/src/tensor_operation_instance/gpu/image_to_column/device_image_to_column_ndhwgc_3d_instance.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bwroblew) [inline library/src/tensor_operation_instance/gpu/image_to_column/device_image_to_column_ndhwgc_3d_instance.cpp]:
```suggestion
// Copyright (c) 2023, Advanced Micro Devices, Inc. All rights reserved.
```

File: library/src/tensor_operation_instance/gpu/image_to_column/device_image_to_column_nhwgc_2d_instance.cpp [hunk 7]
```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bwroblew) [inline library/src/tensor_operation_instance/gpu/image_to_column/device_image_to_column_nhwgc_2d_instance.cpp]:
```suggestion
// Copyright (c) 2023, Advanced Micro Devices, Inc. All rights reserved.
```

File: library/src/tensor_operation_instance/gpu/image_to_column/device_image_to_column_nwgc_1d_instance.cpp [hunk 8]
```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bwroblew) [inline library/src/tensor_operation_instance/gpu/image_to_column/device_image_to_column_nwgc_1d_instance.cpp]:
```suggestion
// Copyright (c) 2023, Advanced Micro Devices, Inc. All rights reserved.
```

**Review by bwroblew — APPROVED** [review]

**Review by LisaDelaney — APPROVED** [review]

**Pull request merged**